### ISSUE TITLE
dev-ai

### ISSUE THREAD

**Issue by devgupta6** [huggingface/transformers#27365]
### Model description

It a generative artifical intelligence model. I have the architecture ready but I am facing the problem to interagte it with transformer architure. It would be great if you provide me assistance with this.

### Open source status

- [X] The model implementation is available
- [ ] The model weights are available

**Comment by amyeroberts** [comment]
Hi @devgupta6, thanks for opening an issue! 

The easiest and recommended way to make a model available in `transformers` is to add the modeling code directly on the hub: https://huggingface.co/docs/transformers/custom_models. Here is a more general guide on adding models: https://huggingface.co/docs/transformers/add_new_model

**Comment by amyeroberts** [comment]
Could you link to your model on the hub? Without seeing the code I'm not able to see what the issue might be.

**Comment by devgupta6** [comment]
Yes, sure. https://github.com/devgupta6/dev-ai this is the model link

**Comment by amyeroberts** [comment]
@devgupta6 Please read the documentation pages I sent over as these contain all the information you should need. The model at the moment is just a torch implementation and doesn't have any of the necessary adaptations for the transformers library. For example, your model needs to inherit from `PretrainedModel`. We can help with bugs and issues but we can't write your code for you.

**Comment by qgallouedec** [comment]
Hey @devgupta6, sorry for the slow reply, catching up on the backlog.

@amyeroberts's answer above still applies: your model needs to subclass `PreTrainedModel` and follow the custom code guide before `AutoModelForCausalLM.register` / `from_pretrained` will work.

https://huggingface.co/docs/transformers/custom_models

Closing since there's been no follow-up, feel free to re-open if you're still stuck after adapting the model.
